src/tallies/tally.cpp

```diff
@@ -419,17 +439,29 @@ void Tally::set_scores(const vector<std::string>& scores)
   bool energyout_present = energyout_filter_ != C_NONE;
   bool legendre_present = false;
   bool cell_present = false;
+  bool energy_present = false;
   bool cellfrom_present = false;
   bool surface_present = false;
   bool meshsurface_present = false;
+  bool non_cell_energy_present = false;
   for (auto i_filt : filters_) {
     const auto* filt {model::tally_filters[i_filt].get()};
+    // Checking for only cell and energy filters for pulse-height tally
+    if (!(filt->type() == FilterType::CELL ||
+          filt->type() == FilterType::ENERGY) ||
+        (filt->type() == FilterType::CELLBORN ||
+          filt->type() == FilterType::CELLFROM) ||
+        filt->type() == FilterType::ENERGY_OUT) {
```

Comment:
I think just checking `!(CELL || ENERGY)` should be sufficient

Reply:
okay, I think I understood the FilterType wrong. This is actually checking the exact type of the filter there is no confusion with Cell and CellBorn for example.